libnestutil/lockptr.h

```diff
@@ -114,7 +114,7 @@ class lockPTR
     ~PointerObject()
     {
       assert( not locked );
-      if ( ( pointee != NULL ) && deletable && ( not locked ) )
+      if ( ( pointee != NULL ) and deletable and not locked )
```

Comment:
Why is there still a `NULL` here?

Reply:
Fixed all of these cases in a separate PR, see #2499

Reply:
OK.